File: api/src/main/java/io/github/jafarlihi/symposium/model/Post.java
package io.github.jafarlihi.symposium.model;

import lombok.Data;

import javax.persistence.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "posts")
@Data
public class Post {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private Long threadId;
    private Long userId;
    private Long postNumber;
    private String content;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
}

File: api/models/post.go
package models

import (
	"time"
)

type Post struct {
	ID         uint32    `json:"id"`
	UserID     uint32    `json:"userID"`
	ThreadID   uint32    `json:"threadID"`
	PostNumber uint32    `json:"postNumber"`
	Content    string    `json:"content"`
	CreatedAt  time.Time `json:"createdAt"`
	Username   string    `json:"username"`
}

type PostAndThread struct {
	ID               uint32    `json:"id"`
	UserID           uint32    `json:"userID"`
	ThreadID         uint32    `json:"threadID"`
	PostNumber       uint32    `json:"postNumber"`
	Content          string    `json:"content"`
	CreatedAt        time.Time `json:"createdAt"`
	Username         string    `json:"username"`
	ThreadUserID     uint32    `json:"threadUserID"`
	ThreadTitle      string    `json:"threadTitle"`
	ThreadCategoryID uint32    `json:"threadCategoryID"`
	ThreadCreatedAt  time.Time `json:"threadCreatedAt"`
}
